### PR TITLE
adds the domain join automation

### PR DESCRIPTION
**Pull request by ghost**
fixes some issues with the linter
adds the domain join tasks
add default inbound and outbound allow all firewall rules
adds comments throughout

### CODE REVIEW

File: tasks/windows.yml [hunk 1]
```diff
@@ -6,11 +7,55 @@
     type: string
     state: present
 
-- name: changing hostname to {{ inventory_hostname }}
+# firewall configuration
```
Comment (bryanlalexander) [inline tasks/windows.yml:10]: Do we actually need to use the host-based firewall?

**Review by bryanlalexander — APPROVED** [review]

File: tasks/windows.yml (hunk 1 above)
Comment (ghost) [reply tasks/windows.yml:10]: I think @juanpgomez-gsa was saying it checks a few compliancy boxes to have it enabled... We won't be doing anything with it so thats why I went all:all in/out to check those boxes.

Comment (bryanlalexander) [reply tasks/windows.yml:10]: ok, well go with it for now we'll circle back on that with security

Comment (ghost) [reply tasks/windows.yml:10]: 👍

**Pull request merged**